### ISSUE TITLE
`crop()` promise resolved with empty object 

### ISSUE THREAD

**Issue by tamakisquare** [JrSchild/jr-crop#25]
Device: iPhone 6
Platform: iOS 9.1
jr-crop version: 1.1.2

For my setup, the image for cropping is a feed from **cordova-plugin-camera** (via **ngCordova**), either as a photo selected from the library or a new photo taken at the time. The image shows up fine on the cropping screen (it's even using the new `circle` feature). The promise returned from `crop()`, however, is always *resolved* with empty object, no matter how I pan and/or pinch the circular cropping window. 

My code (below) is very straightforward and I have spent hours going through it, so I am quite confident with it. I looked through all the issues (*open* and *closed*) and was surprised that no one had encountered the same issue, except for https://github.com/JrSchild/jr-crop/issues/16. But that issue was reported with vague description and no deep analysis on the problem, so it's difficult to tell whether that's of the same issue as what I am reporting here.

From going through all the issues in this repo, I've got this impression that most of the issues with *cordova-plugin-camera* involved, if not all, are using `Camera.DestinationType.DATA_URL` rather than `Camera.DestinationType.FILE_URI`. The former works with *base64-encoded string*, while the latter deals with **file url** (in iOS, it's *"file:///..."*).

Based on the observations above, I have a feeling that this issue has something to do with feeding "jr-crop" with *file url*.

It would be great if someone who has worked with the `Camera.DestinationType.FILE_URI` + "jr-crop" combination, regardless working or not, can share his/her experience with me. Please state the device and o/s version you used at the time.

```javascript
var nativeCameraOpts = {
  quality: 90,
  targetWidth: 2000,
  targetHeight: 2000,
  destinationType: Camera.DestinationType.FILE_URI,
  encodingType: Camera.EncodingType.JPEG,
  allowEdit: false,
  mediaType: Camera.MediaType.PICTURE,
  saveToPhotoAlbum: false,
  correctOrientation: true
};

// `$cordovaCamera` is a wrapper for `cordova-plugin-camera` provided by `ngCordova` module.
$cordovaCamera.getPicture(nativeCameraOpts).then(getPictureSucc);

function getPictureSucc(imageURI) {
  // An example of `imageURI` is "file:///var/mobile/Containers/Data/Application/F0C5EC0D-23EA-4D43-83BE-A850FA89FCEC/tmp/cdv_photo_005.jpg"
  console.log('Picture source URI: ' + imageURI);  

  // For debugging, to make sure the image coming from `cordova-plugin-camera` is valid.
  angular.element('<img />')
    .bind('load', function(e) {
      console.log('Width: ' + this.naturalWidth);
      console.log('Height: ' + this.naturalHeight);
    })
    .prop('src', imageURI);

  $jrCrop.crop({
    url: imageURI,
    width: 320,
    height: 320,
    title: 'Move and Scale',
    circle: true
  }).then(cropPictureSucc);

  function cropPictureSucc(cropped) {
    console.log(cropped); // `cropped` is `{}`

    var dataUrl = cropped.toDataUrl();  // `toDataUrl` does not exist because `cropped` is an empty object.

    // ... send `dataUrl` to the server.
  }
}
```

**Comment by JrSchild** [comment]
Hi @tamakisquare. Thank you for opening an issue. I quickly tested your example and it seems to work in the simulator with `sourceType: Camera.PictureSourceType.PHOTOLIBRARY`. The promise is always resolved with a canvas (an empty object or undefined is unlikely). The only error I see so far is a typo.

Did you try `cropped.toDataURL()` instead of `cropped.toDataUrl()`?

**Comment by tamakisquare** [comment]
Hi @JrSchild, thanks for taking a look of this issue and pointing out my syntax error. 

After correcting that syntax error, I tried the code in the iOS simulator but I got the same result (ie. promise got resolved to an empty object). 

My setup is 
Simulator: iPhone 6 / iOS 9.1
Xcode: 7.1.1 (7B1005)

May I ask what was the setup you used for the emulation?

If there's anything else you can think of to investigate this case further, don't hesitate to let me know and I'll see what I can do. Meanwhile, for my project, I am going to switch to using `DATA_URL`.

**Comment by tamakisquare** [comment]
@JrSchild , FYI. After I switched over to using `DATA_URL` instead `FILE_URI`, the cropping started to work as expected.

I suspect the issue has something to do with some security mechanism at some layer in the setup mentioned in my previous comment. So far, I have only checked `imgSrcSanitizationWhitelist` in AngularJS and, by default, it has already been configured to accept *"file:///..."*, so that's not the culprit.

**Comment by JrSchild** [comment]
Hi Tamaki,

[This was the example](https://github.com/JrSchild/jr-crop-cordova) that I used with `FILE_URI`, see README on how to get it running. Also note that I have updated the [examples](https://github.com/JrSchild/jr-crop/commit/703837f8ebf5eb77bee36180fe89464f5442532b) folder with how to create a BLOB and upload + store the image on the server.

Closing, assuming it's working, feel free to continue the discussion below or re-open the issue if problems remain.
